contracts/vault/src/execute.rs

```diff
@@ -0,0 +1,591 @@
+use perpswap::contracts::vault::ExecuteMsg;
+
+use crate::{
+    common::{check_not_paused, get_total_assets, is_authorized},
+    prelude::*,
+    state,
+};
+
+// External interfaces (simulated for this example)
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum FactoryQueryMsg {
```

Comment:
We have access to the factory query messages in the perpswap crate, any reason you're not using those directly?